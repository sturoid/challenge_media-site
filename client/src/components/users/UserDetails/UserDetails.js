import React from 'react';
import userDetails from './UserDetails.styles';
import { userDataFieldType } from '../../../lib/propTypes';

const UserDetails = ({ user }) => (
  <div className={userDetails}>
    <div>{`${user.first_name} ${user.last_name}`}</div>
    <div>{`${user.state}, ${user.country}`}</div>
  </div>
);

UserDetails.propTypes = {
  user: userDataFieldType.isRequired
};

export default UserDetails;
