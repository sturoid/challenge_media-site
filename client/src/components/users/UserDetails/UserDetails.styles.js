import { css } from 'emotion';

const userDetails = css`
  & > div:first-child {
    padding: 20px 0;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  & > div:not(first-child) {
    padding: 5px 0;
  }
`;

export default userDetails;
