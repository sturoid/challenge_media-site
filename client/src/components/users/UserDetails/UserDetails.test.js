import React from 'react';
import renderer from 'react-test-renderer';
import UserDetails from './UserDetails';
import userData from '../../../../../lib/test-data/user';

describe('Rendering', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer.create(<UserDetails user={userData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
