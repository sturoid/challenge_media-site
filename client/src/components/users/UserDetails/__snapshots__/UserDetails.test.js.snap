// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetails /> renders correctly and equals snapshot 1`] = `
<div
  className="css-1rypxo3"
>
  <div>
    Johny vinoâ„¢ .
  </div>
  <div>
    New York, United States
  </div>
</div>
`;
